### PR TITLE
Add `currently_in_test_context()` to check if you can call `assume()`, `target()`, etc.

### LINKED ISSUE #2581: target() cannot be called outside a hypothesis test
I have a test `assert_allclose_phase` which checks whether two arrays are almost equal modulo one. When running under `hypothesis`, this seems like a sensible place to put `target`, albeit with a(n optional?) `name` argument in case I want to call it twice in the same test. Unfortunately if I do so then I can't use the same function in non-`hypothesis` tests:

```
E           hypothesis.errors.InvalidArgument: Calling target() outside of a test is invalid.
```

My current workaround is to call `target` only if `name` is provided, but this clutters simple tests that wouldn't need a `name`.

### PR DESCRIPTION
**Pull request by Zac-HD**
Fixes #2581, taking the approach that @PiDelport suggested in #2675.

### CODE REVIEW

File: hypothesis-python/RELEASE.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+RELEASE_TYPE: minor
+
+This release adds :func:`hypothesis.currently_in_test_context`, which can be used
+to check whether the calling code is currently running inside an
```
Comment (PiDelport) [inline hypothesis-python/RELEASE.rst:4]: an → a ?

(unless this is intended to go with a pronounced `@`?)

Comment (PiDelport) [reply hypothesis-python/RELEASE.rst:4]: (Ditto for the docstring.)

**Review by PiDelport — APPROVED** [review]
The name sounds good to me!

File: hypothesis-python/RELEASE.rst (hunk 1 above)
Comment (Zac-HD) [reply hypothesis-python/RELEASE.rst:4]: I tend to write "an", mentally pronouncing "an at-given", but a quick grep through our docs reveals that we use both... and I'm going to leave that as a problem for another day 😅

Comment (PiDelport) [reply hypothesis-python/RELEASE.rst:4]: English! How does it even? 😄️

**Pull request merged**